libcaf_core/caf/system_messages.test.cpp

```diff
@@ -0,0 +1,257 @@
+    check_ne(msg3, msg1);
+  }
+}
+
+TEST("down_msg is serializable") {
+  SECTION("invalid actor address") {
+    auto msg1 = down_msg{nullptr, sec::runtime_error};
+    auto msg2 = serialization_roundtrip(msg1);
+    check_eq(msg1, msg2);
+  }
+  SECTION("valid actor address") {
+    // Note: serializing an actor puts it into the registry. Hence, we need to
+    //       shut down the actor manually before the end because its reference
+    //       count will not drop to zero otherwise.
+    auto dummy = sys.spawn([] { return behavior{[](int) {}}; });
+    auto dummy_guard = make_actor_scope_guard(dummy);
+    auto msg1 = exit_msg{dummy.address(), sec::runtime_error};
+    auto msg2 = serialization_roundtrip(msg1);
+    check_eq(msg1, msg2);
+  }
+}
+
+TEST("group_down_msg is serializable") {
+  auto msg1 = group_down_msg{sys.groups().anonymous()};
+  auto msg2 = serialization_roundtrip(msg1);
+  if (check(msg2.has_value()))
+    check_eq(msg1.source, msg2.value().source);
+}
+
+TEST("node_down_msg is comparable") {
+  SECTION("empty node") {
+    auto msg1 = node_down_msg{node_id{}, sec::runtime_error};
+    auto msg2 = node_down_msg{node_id{}, sec::runtime_error};
+    auto msg3 = node_down_msg{node_id{}, sec::disposed};
+    check_eq(msg1, msg2);
+    check_eq(msg2, msg1);
+    check_ne(msg1, msg3);
+  }
+  SECTION("node from hashed_node_id") {
+    auto msg1 = node_down_msg{node_id{hashed_node_id{}}, sec::runtime_error};
+    auto msg2 = node_down_msg{node_id{hashed_node_id{}}, sec::runtime_error};
+    auto msg3 = node_down_msg{node_id{hashed_node_id{}}, sec::disposed};
+    check_eq(msg1, msg2);
+    check_eq(msg2, msg1);
+    check_ne(msg1, msg3);
+  }
+  SECTION("node from uri") {
+    auto msg1 = node_down_msg{node_id{uri{}}, sec::runtime_error};
+    auto msg2 = node_down_msg{node_id{uri{}}, sec::runtime_error};
+    auto msg3 = node_down_msg{node_id{uri{}}, sec::disposed};
+    check_eq(msg1, msg2);
+    check_eq(msg2, msg1);
+    check_ne(msg1, msg3);
+  }
+}
+
+TEST("node_down_msg is serializable") {
+  SECTION("empty node") {
+    auto msg1 = node_down_msg{node_id{}, sec::runtime_error};
+    auto msg2 = serialization_roundtrip(msg1);
+    check_eq(msg1, msg2);
+  }
+  SECTION("node from hashed_node_id") {
+    auto msg1 = node_down_msg{node_id{hashed_node_id{}}, sec::runtime_error};
+    auto msg2 = serialization_roundtrip(msg1);
+    check_eq(msg1, msg2);
+  }
+  SECTION("node from uri") {
+    auto msg1 = node_down_msg{node_id{uri{}}, sec::runtime_error};
+    auto msg2 = serialization_roundtrip(msg1);
+    check_eq(msg1, msg2);
+  }
+}
+
+TEST("stream_open_msg is serializable") {
+  SECTION("invalid strong_actor_ptr") {
+    auto msg1 = stream_open_msg{42, nullptr, 43};
+    auto msg2 = serialization_roundtrip(msg1);
+    if (check(msg2.has_value())) {
+      check_eq(msg1.id, msg2.value().id);
+      check_eq(msg1.sink, msg2.value().sink);
+      check_eq(msg1.sink_flow_id, msg2.value().sink_flow_id);
```

Comment:
```suggestion
      check_eq(msg1.id, msg2->id);
      check_eq(msg1.sink, msg2->sink);
      check_eq(msg1.sink_flow_id, msg2->sink_flow_id);
```

Same for the other uses of `.value().` down below.